src/benchmark/file_io_write_micro_benchmark.cpp

```diff
@@ -14,19 +14,48 @@ const int32_t MB = 1000000;
 
 class FileIOWriteMicroBenchmarkFixture : public MicroBenchmarkBasicFixture {
  public:
+  uint64_t control_sum = uint64_t{0};
+
   void SetUp(::benchmark::State& state) override {
     // TODO(phoeinx): Make setup/teardown global per file size to improve benchmark speed
     ssize_t BUFFER_SIZE_MB = state.range(0);
     // each int32_t contains four bytes
     int32_t vector_element_count = (BUFFER_SIZE_MB * MB) / 4;
     data_to_write = std::vector<int32_t>(vector_element_count, 42);
```

Comment:
not part of your code, but perhaps we shouldn't write exactly the same element over and over? Not sure if the OS optimizes on pages with always the same element. (It for sure does for 0-pages.)

Reply:
Lets put this to a later point in time and first merge this PR to get the code in.